### PR TITLE
Update 2.0.1

### PR DESCRIPTION
**Pull request by SC426**
[Upstream](https://github.com/shapely/shapely)
[Changelog](https://github.com/shapely/shapely/blob/main/CHANGES.txt)

### Actions
- Update to version 2.0.1
- Update dependencies
- Remove outdated build & test files (build.sh, bld.bat, run_test.py)
- Linter fixes

### Notes
- `build.sh`, `bld.bat`, and `run_test.py` are no longer required.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: a195e51caafa218291f2cbaa3fef69fd3353c93ec4b65b2a4722c4cf40c3198c
+  url: https://github.com/{{ name }}/{{ name }}/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 6250e79ca299222a239ce44df613d228f1216c5cc31351217bf377b03d7be4e0
 
 build:
   number: 0
   # geos currently isn't available on s390x
-  skip: True  # [py<36 or s390x]
+  skip: True  # [py<37 or s390x]
```
Comment (ELundby45) [inline recipe/meta.yaml:15]:
```suggestion
  skip: True  # [py<37 or s390x]
  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
```

**Review by ELundby45 — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -36,14 +34,9 @@ test:
...
-    - pytest
   imports:
-    - shapely
-    - shapely.algorithms
-    - shapely.examples
-    - shapely.geometry
-    - shapely.speedups
-    - shapely.vectorized
+    # ModuleNotFoundError: No module named 'shapely'
+    # - shapely
```
Comment (ELundby45) [inline recipe/meta.yaml]: We should be able to add most or all of these back after `script` is added to the build section

File: recipe/meta.yaml [hunk 3]
```diff
@@ -52,9 +45,8 @@ about:
   license: BSD-3-Clause
   license_family: BSD
   license_file: LICENSE.txt
-  license_url: https://github.com/shapely/shapely/blob/main/LICENSE.txt
   summary: "Python package for manipulation and analysis of geometric objects in the Cartesian plane"
-
+  description: Shapely is a BSD-licensed Python package for manipulation and analysis of planar geometric objects.
   dev_url: https://github.com/Toblerity/Shapely
```
Comment (ELundby45) [inline recipe/meta.yaml]:
```suggestion
  dev_url: https://github.com/shapely/shapely
```

File: recipe/meta.yaml (hunk 1 above)
Comment (SC426) [reply recipe/meta.yaml:15]: Got a new error after adding the script: 
`Building wheel for shapely (pyproject.toml): finished with status 'error'
  ERROR: Failed building wheel for shapely
Failed to build shapely
ERROR: Could not build wheels for shapely, which is required to install pyproject.toml-based projects`

File: recipe/meta.yaml [hunk 4]
```diff
@@ -40,22 +42,20 @@ test:
...
-    - shapely.examples
     - shapely.geometry
-    - shapely.speedups
     - shapely.vectorized
   commands:
     - pip check
+    - pytest --pyargs shapely
 
 about:
   home: https://github.com/Toblerity/Shapely
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  home: https://github.com/shapely/shapely
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,31 +1,33 @@
...
   build:
     - {{ compiler('c') }}
   host:
     - python
-    - cython >=0.29.24,<3
-    - geos
-    - numpy   1.19  # [py<310]
-    - numpy   1.21  # [py>=310]
+    - cython 0.29.33
+    - numpy
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - numpy {{ numpy }}
```

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,31 +1,33 @@
...
+    - geos 3.9.1
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - geos {{ geos }}
```

Reason: geos is pinned in aggregate conda_build_config as this:
```
geos:
  - 3.8.0  # [not (osx and arm64)]
  - 3.9.1  # [osx and arm64]
```

Comment (SC426) [reply recipe/meta.yaml]: Thanks for that info!

**Review by ELundby45 — APPROVED** [review]
LGTM

**Review by cbouss — APPROVED** [review]

**Pull request merged**